File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cpp
#include <bits/stdc++.h>

using namespace std;

bool checkIsBabbling(string word) {
    for (char w : word) {
        if (!isdigit(w)) {
            return false;
        }
    }
    
    return true;
}

int solution(vector<string> babbling) {
    vector<string> words = {"aya", "ye", "woo", "ma"};
    
    for (string& b : babbling) {
        for (string w : words) {
            auto idx = b.find(w);
            
            if (idx != string::npos) {
                for (auto i = idx; i < idx + w.size(); i++){
                    b[i] = '1';
                }
            }
        }
    }
        
    int answer = 0;
    
    for (string b: babbling) {
        if(checkIsBabbling(b)) {
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    String[] spells = {"aya", "ye", "woo", "ma"};
    
    public int solution(String[] babbling) {
        int answer = 0;
        
        // 1. 주어진 옹알이에 네 가지 발음이 포함되어 있는지 확인 후, 대체
        // 각 문자열에 발음은 최대 한 번만 들어있다.
        for (int i = 0; i < babbling.length; i++) {
            for (String spell : spells) {
                babbling[i] = babbling[i].replace(spell, "1");
            }
            babbling[i] = babbling[i].replace("1", "");
            
            // 2. 옹알이 단어가 비어있다면 카운트
            if (babbling[i].isEmpty()) answer++;
        }

        return answer;
    }
}